packages/plugins/src/foundry/index.ts

```diff
@@ -36,6 +45,59 @@ import { createChugSplashRuntime } from '../utils'
 const args = process.argv.slice(2)
 const command = args[0]
 
+const decodeCachedConfig = (encodedConfigCache: string) => {
+  const ChugSplashFoundryABI =
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    require('../../out/artifacts/ChugSplash.sol/ChugSplash.json').abi
+  const configCacheType = ChugSplashFoundryABI.find(
+    (fragment) => fragment.name === 'getConfigCache'
+  ).outputs[0]
+
+  const configCache = ethers.utils.defaultAbiCoder.decode(
+    [configCacheType],
+    encodedConfigCache
+  )[0]
+
+  const structuredConfigCache = {
+    blockGasLimit: configCache.blockGasLimit,
+    liveNetwork: configCache.liveNetwork,
+    networkName: configCache.networkName,
+    contractConfigCache: {},
+  }
+
+  for (const cachedContract of configCache.contractConfigCache) {
+    structuredConfigCache.contractConfigCache[cachedContract.referenceName] = {
+      referenceName: cachedContract.referenceName,
+      isTargetDeployed: cachedContract.isTargetDeployed,
+      deploymentRevert: {
```

Comment:
This is probably the jankiest part of this PR. Here I am taking the ConfigCache object which has been encoded as a bytes string, and decoding it using the Struct ABI. I then do some modifications to it to get it into the expected format for our Typescript logic. This is required because the ConfigCache object type is slightly different from the typescript version.

So yeah, this could end up being annoying to maintain b/c if we ever need to change this object it'll require updating this logic to make sure it's properly translated. We could probably simplify this by removing the `Optional<Type>` pattern in favor of using expected defaults. I.e an undefined address is just `address(0)` and then our typescript logic can detect that and handle it. This is just an idea though, I don't have a particularly strong opinion if it's necessary or not.

Reply:
Ah i see. I'll reconsider the struct design. LGTM though